site/src/pages/GroupsPage/GroupPage.tsx

```diff
@@ -246,15 +230,20 @@ export const GroupPage: React.FC = () => {
 
       {group && (
         <DeleteDialog
-          isOpen={state.matches("confirmingDelete")}
-          confirmLoading={state.matches("deleting")}
+          isOpen={isDeletingGroup}
+          confirmLoading={deleteGroupMutation.isLoading}
           name={group.name}
           entity="group"
-          onConfirm={() => {
-            send("CONFIRM_DELETE");
+          onConfirm={async () => {
+            try {
+              await deleteGroupMutation.mutateAsync(groupId);
+              navigate("/groups");
+            } catch (error) {
+              displayError(getErrorMessage(error, "Failed to delete group."));
+            }
           }}
           onCancel={() => {
```

Comment:
Clarifying question: the `onCancel` callback is expected just to close out UI elements, instead of needing to interface with the server?

Reply:
Yeap.